backend/lib/geocoder.py

```diff
@@ -61,23 +63,30 @@ def geocode_batch(self, addresses):
                 if results.get(address) and results.get(address).coords
             ]
         except Exception as error:
-            print(error.__class__, error, 'in batch geocoding - switching to single geocoding.')
+            print(error.__class__, error, 'in batch geocoding. Switching to single geocoding.')
 
             with concurrent.futures.ThreadPoolExecutor(max_workers=MAX_THREADS) as executor:
                 geocoded_addresses = executor.map(self.geocode, addresses)
 
-            return [
-                geocoded_address for geocoded_address
-                in geocoded_addresses if geocoded_address
-            ]
+        return [
+            geocoded_address for geocoded_address
+            in geocoded_addresses if geocoded_address
+        ]
 
     @retry(
-        retry_on_exception=_retry_on_geocodio_errors,
+        retry_on_result=retry_if_result_none,
         wait_fixed=WAIT_FIXED_MILLISECONDS,
         stop_max_attempt_number=STOP_MAX_ATTEMPT_NUMBER)
     def _safe_geocode(self, addresses):
         # TODO: Handle more than 10,000 addresses at once.
-        return self.client.geocode(addresses)
+        try:
+            return self.client.geocode(addresses)
+        except (GeocodioAuthError, GeocodioDataError, GeocodioServerError, GeocodioError) as error:
+            if isinstance(addresses, list):
+                print(error, 'Geocoding address(es) {}.'.format(addresses))
```

Comment:
are you sure we want to print all of the addresses?

Reply:
Good catch